types/react-router/index.d.ts

```diff
@@ -55,12 +55,12 @@ declare module 'react-router' {
 
   interface RouteComponentProps<P> {
     match: match<P>;
-    location: H.Location;
+    location: H.LocationDescriptor;
```

Comment:
This change can pose a problem because the [major use case for the `Route` component is via a `component` prop](https://reacttraining.com/react-router/core/api/Route/Route-render-methods). 

Suppose I declare this route config:
```jsx
<Router>
    <Route path='/some/route' component={Child}/>
</Router>
```
and then `Child` uses the `location` provided by the `Router` in some way. A natural way to type `Child` is have its `props` have the type of `RouteComponentProps`
```jsx
const Child = (props: RouteComponentProps<params>) => (
    <div>props.location.pathname</div>
)
```
because that's what [`react-router` passes to `Child`](https://github.com/ReactTraining/react-router/blob/61dc090fa8ac349508fa0f3345c1365dd5638a32/packages/react-router/modules/Route.js#L105). If this change is accepted then TypeScript will complain that `pathname` may not exist because all of `location`'s properties are optional (because its type is `LocationDescriptor`).

@elmariofredo do you have an example that requires changing the type to `LocationDescriptor`?